### ISSUE TITLE
Unable to find a constructor to use for type MatrixServerVersionsResponse

### ISSUE THREAD

**Issue by fraidev** [trilitech/tezos-unity-sdk#90]
When using the SDK on iOS, I get a JSON serialization error. 

```
JsonSerializationException: Unable to find a constructor to use for type Matrix.Sdk.Core.Infrastructure.Dto.ClientVersion.MatrixServerVersionsResponse. A class should either have a default constructor, one constructor with arguments or a constructor marked with the JsonConstructor attribute. Path 'versions', line 1, position 12.
```

I **think** this is the reason because iOS authentication is buggy.

~I think that we need to put a default constructor here:
[Matrix.Sdk/Core/Infrastructure/Dto/ClientVersion/MatrixServerVersionsResponse.cs](https://github.com/baking-bad/matrix-dotnet-sdk/blob/main/Matrix.Sdk/Core/Infrastructure/Dto/ClientVersion/MatrixServerVersionsResponse.cs)~

Edit:
Actually, changing Managed Stripping Level from Low to Minimal fix it!
![image](https://github.com/trilitech/tezos-unity-sdk/assets/25258368/4e7d9f0e-7397-45ca-90b3-ab5b28e6b502)

Maybe we should add a [link.xml](https://stackoverflow.com/a/59486989/8778072) in the sample folder to force it or write in the README that only works on iOS with Managed Stripping Level selected to Minimal.

**Comment by fraidev** [comment]
I added a PR that will fix it for imported samples #91.

But this will still happen for how only use SDK without importing samples.

I think that we need to improve documentation about it or create a preprocessor to do it:
https://forum.unity.com/threads/the-current-state-of-link-xml-in-packages.995848/#post-7223887

**Comment by k-karuna** [comment]
@fraidev thanks for investigation. By default in my Unity ver (2021.3.27f) is set Managed Stripping Level by default to Disabled, thereby i haven't faced this issue. But when i switch to any different value, i'm receiving following error during building Android
```
Fatal error in Unity CIL Linker
Mono.Cecil.AssemblyResolutionException: Failed to resolve assembly: 'Newtonsoft.Json, Version=13.0.0.0.
```
So, i would to suggest update Readme to disable Managed Stripping Level on mobile platforms.
